tests/test_async_utils.py

```diff
@@ -164,3 +165,38 @@ class TestException(Exception):
 
     # should not raise a KeyError!
     await asyncio.gather(awaiter(), awaiter())
+
+
+@pytest.mark.asyncio
+async def test_async_ttl_cache_for_class_members_doesnt_leak_mem():
+    """Ensure that we aren't leaking memory"""
+
+    x = 42
+
+    class TestClass:
+        @async_ttl_cache(ttl=None, cleanup_self=True)
+        async def f(self):
+            return x
+
+    # `instance_caches` from `async_ttl_cache()` applied to `TestClass.f`.
+    instance_caches = TestClass.f.__closure__[1].cell_contents
```

Comment:
Suggestion: I added a `cache` kwarg to `async_ttl_cache` so that I wouldn't have to do this.  You could use that instead to inject the cache instead of having to pull it out of the closure.